### PR TITLE
Code & Comment updates on core contracts

### PR DESCRIPTION
**Pull request by codenamejason**
Included:
- updates if statements with reverts to one line when formatting allows it
- adds/removes/updates some comments and NATSPEC

Example:
```typescript
if (_owner == address(0)) {
    revert ZERO_ADDRESS();
}
```

is now

```typescript
if (_owner == address(0)) revert ZERO_ADDRESS();
```

This reduces the number of lines of code as well as make it look a bit cleaner.

### CODE REVIEW

**Review by Aryan9592 — APPROVED** [review]
Nice add!

File: contracts/strategies/qv-hackathon/SchemaResolver.sol [hunk 1]
```diff
@@ -13,9 +12,7 @@ import {IEAS, Attestation} from "eas-contracts/IEAS.sol";
 import {InvalidEAS, uncheckedInc} from "eas-contracts/Common.sol";
 import {ISchemaResolver} from "eas-contracts/resolver/ISchemaResolver.sol";
 
-/**
- * @title A base resolver contract
- */
+/// @title A base resolver contract
```
Comment (thelostone-mc) [inline contracts/strategies/qv-hackathon/SchemaResolver.sol:15]: would recommend not changing anythig on this file as we've copied it.

File: contracts/strategies/qv-base/QVBaseStrategy.sol [hunk 2]
```diff
@@ -257,14 +257,14 @@ abstract contract QVBaseStrategy is BaseStrategy {
...
     {
-        uint256 recipientLength = _recipientIds.length;
-        if (recipientLength != _recipientStatuses.length) {
-            revert INVALID();
-        }
+        // make sure the arrays are the same length
+        if (_recipientIds.length != _recipientStatuses.length) revert INVALID();
 
-        for (uint256 i; i < recipientLength;) {
+        for (uint256 i; i < _recipientIds.length;) {
```
Comment (thelostone-mc) [inline contracts/strategies/qv-base/QVBaseStrategy.sol:263]: this change increases gas as the .length would have to be computed at ever iteration

**Review by thelostone-mc — CHANGES_REQUESTED** [review]
I'd recommend we stick to wrapping if statements within a curly brackets cause it's less error prone https://softwareengineering.stackexchange.com/questions/16528/single-statement-if-block-braces-or-no

File: contracts/core/Allo.sol [hunk 3]
```diff
@@ -9,6 +9,7 @@ import "openzeppelin-contracts/contracts/access/AccessControl.sol";
 import "openzeppelin-contracts-upgradeable/contracts/security/ReentrancyGuardUpgradeable.sol";
 // Interfaces
 import "./interfaces/IAllo.sol";
+import "./interfaces/IRegistry.sol";
```
Comment (thelostone-mc) [inline contracts/core/Allo.sol:12]:
```suggestion
```

File: contracts/strategies/BaseStrategy.sol [hunk 4]
```diff
@@ -214,12 +213,11 @@ abstract contract BaseStrategy is IStrategy, Transfer, Errors {
...
-        if (length != _data.length) {
-            revert ARRAY_MISMATCH();
-        }
-        PayoutSummary[] memory payouts = new PayoutSummary[](length);
-        for (uint256 i; i < length;) {
+        // check if the length of the recipient IDs and data arrays are equal, if they are not, revert
+        if (_recipientIds.length != _data.length) revert ARRAY_MISMATCH();
+
+        PayoutSummary[] memory payouts = new PayoutSummary[](_recipientIds.length);
+        for (uint256 i; i < _recipientIds.length;) {
```
Comment (thelostone-mc) [inline contracts/strategies/BaseStrategy.sol:220]: uint256 length = _recipientIds.length;

**Review by thelostone-mc — APPROVED** [review]

**Review by 0xKurt — APPROVED** [review]

**Pull request merged**